service/handlers.go

```diff
@@ -113,3 +125,47 @@ func (e *ExecuteStruct) GetClient(projectID string) (client.RancherClient, error
 	}
 	return *apiClient, nil
 }
+
+func saveWebhook(webhookRequestData map[string]interface{}) error {
+	config := config.GetConfig()
+	if _, ok := webhookRequestData["name"].(string); !ok {
+		return fmt.Errorf("Name not found after decode")
+	}
+	webhookName := webhookRequestData["name"].(string)
+	apiClient, err := client.NewRancherClient(&client.ClientOpts{
+		Timeout: time.Second * 30,
+		Url:     config.CattleURL,
+	})
+
+	if err != nil {
+		return fmt.Errorf("Failed to create client")
+	}
+
+	UUID := webhookRequestData["uuid"].(string)
+
+	_, err = apiClient.Webhook.Create(&client.Webhook{
+		Name: webhookName,
+		Key:  UUID,
+	})
+
+	if err != nil {
+		return fmt.Errorf("Failed to create webhook : %v", err)
+	}
+	return nil
+}
+
+func validateWebhook(claims map[string]interface{}, apiClient client.RancherClient) error {
```

Comment:
Make the return signature (int, error) where int is an http response code to return. Return a 403 if the webhook has been revoked. Return a 500 if there is an error calling the rancher API. In the calling function, use that int as the response code.